### DISCUSSION TITLE
how use bigdecimal with serde?

### DISCUSSION THREAD

**Discussion by traitmeta** [SeaQL/sea-orm#1926] [Q&A] [answered]
There will be errors if I use bigDecimal features and add serde in derive. I found bigDecimal code can choose serde.  so how can I do it correctly？

```
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, Serialize, Deserialize)]
#[sea_orm(table_name = "token")]
pub struct Model {
    #[sea_orm(
        primary_key,
        auto_increment = false,
        column_type = "Binary(BlobSize::Blob(None))"
    )]
    pub hash: Vec<u8>,
    pub amount: Option<BigDecimal>,
}
```

**Comment by tyt2y3** [comment]
I think the rule-of-thumb is always serde Decimal or BigDecimal as string, I couldn't remember if there is a feature flag for that with the libraries, but you can always use a custom serde attribute `#[serde(with = "big_decimal_to_string")]`

**Reply by traitmeta** [reply, marked as answer]
thanks, I add `bigdecimal = { version = "0.3", features = ["serde"] }` in dependencies and than it's work.
